cassiopeia/solver/SpectralNeighborJoiningSolver.py

```diff
@@ -0,0 +1,281 @@
+"""
+This file stores a subclass of DistanceSolver, SpectralNeighborJoining. This algorithm is based on the one developed by Jaffe, et al. (2020) in their paper titled "Spectral neighbor joining for reconstruction of latent tree models. 
+"""
+
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import abc
+import networkx as nx
+import numpy as np
+import pandas as pd
+from itertools import chain
+
+# Only for debugging purposes
+# if True:
+#     import sys
+#     sys.path.insert(0, '../..')
+
+from cassiopeia.solver.DistanceSolver import DistanceSolver, DistanceSolverError
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.solver import (
+    dissimilarity_functions,
+    solver_utilities,
+)
+
+from sklearn.decomposition import TruncatedSVD
+from scipy.sparse.linalg import svds, eigs
+from scipy.linalg import svd as scipy_svd
+
+
+class SpectralNeighborJoiningSolver(DistanceSolver):
+    """
+    Spectral Neighbor-joining class for Cassiopeia.
+
+    Implements a variation on the Spectral Neighbor-Joining algorithm described by Jaffe et al. in their 2020 paper. This class implements the abstract class DistanceSolver. This class inherits the 'solve' method from DistanceSolver with the find_cherry method implemented using a spectral method as in Jaffe et al. (2020).
+
+    Args:
```

Comment:
so I put the default for dissimilarity_map ('exponential_negative_hamming_distance') as a method in the dissimilarity_functions.py

Would it be weird to put a similarity map in the dissimilarity_functions.py?

I was also thinking of just taking in dissimilarity maps and transforming it into a similarity one when we actually use it but this is probably a terrible idea...?

Reply:
so I tried changing replacing the places where we use "dissimilarity" with similarity, but because the abstract class DistanceSolver uses methods that are called get_dissimilarity_map or update_dissimilarity_map, when i rename them to similarity it errors out. Is there a cleaner way to do this? should i rename the methods in the DistanceSolver class? somethign like "affinity" perhaps?

Reply:
nvm we'd need to change a bunch of files if we did this